src/kv/test/kv_serialisation.cpp

```diff
@@ -88,30 +61,87 @@ TEST_CASE(
 
   INFO("Deserialise transaction in target store");
   {
+    REQUIRE(consensus->get_latest_data().second);
+    REQUIRE(!consensus->get_latest_data().first.empty());
     REQUIRE(
       kv_store_target.deserialise(consensus->get_latest_data().first) ==
       kv::DeserialiseSuccess::PASS);
 
     kv::Tx tx_target;
-    auto view_target = tx_target.get_view(
-      *kv_store_target.get<std::string, std::string>("pub_map"));
+    auto view_target = tx_target.get_view(*target_map);
     REQUIRE(view_target->get("pubk1") == "pubv1");
   }
 }
 
-TEST_CASE(
+TEST_CASE("Comparison")
+{
+  std::vector<uint8_t> raw_data;
+  {
+    auto consensus = std::make_shared<kv::StubConsensus>();
+    kv::Store kv_store(consensus);
+
+    auto& map0 = kv_store.create<RawMapTypes::StringString>(
+      "map0", kv::SecurityDomain::PUBLIC);
+    auto& map1 =
+      kv_store.create<RawMapTypes::NumNum>("map1", kv::SecurityDomain::PUBLIC);
+
+    kv::Tx tx;
+    auto view0 = tx.get_view(map0);
+    view0->put("pubk1", "pubv1");
+    auto view1 = tx.get_view(map1);
+    view1->put(0xab, 0xef);
+    REQUIRE(tx.commit() == kv::CommitSuccess::OK);
+    raw_data = consensus->get_latest_data().first;
+  }
+
+  std::vector<uint8_t> exp_data;
+  {
+    auto consensus = std::make_shared<kv::StubConsensus>();
+
+    kv::Store kv_store(consensus);
+
+    auto& map0 = kv_store.create<ExperimentalMapTypes::StringString>(
+      "map0", kv::SecurityDomain::PUBLIC);
+    auto& map1 = kv_store.create<ExperimentalMapTypes::NumNum>(
+      "map1", kv::SecurityDomain::PUBLIC);
+
+    kv::Tx tx;
+    auto view0 = tx.get_view(map0);
+    view0->put("pubk1", "pubv1");
+    auto view1 = tx.get_view(map1);
+    view1->put(0xab, 0xef);
+    REQUIRE(tx.commit() == kv::CommitSuccess::OK);
+    exp_data = consensus->get_latest_data().first;
+  }
+
+  // The new form is slightly longer but this is inevitable, and acceptable?
+  // TODO: Remove this test
+  // CHECK(raw_data == exp_data);
```

Comment:
I was interested if the 2 implementations produce identical serialisations, but (at least for msgpack) they don't quite. Essentially we create an extra layer of nesting, since the 'raw' serialisation is now contained in a vector, and we include 'type' information in that vector. The difference is just a few bytes, so I think this is fine for now. In future, we can remove this by blitting the serialised Ks/Vs directly, and not calling `serialise_value` etc for them.

For comparison, here's the output of this test, highlighting the additional nested headers:

Old:
01 02 a4 6d 61 70 30 d3 80 00 00 00 00 00 00 00 00 01 a5 70 75 62 6b 31 a5 70 75 62 76 31 00 02 a4 6d 61 70 31 d3 80 00 00 00 00 00 00 00 00 01 cc ab cc ef 00
New:
01 02 a4 6d 61 70 30 d3 80 00 00 00 00 00 00 00 00 01 **c4 06** a5 70 75 62 6b 31 **c4 06** a5 70 75 62 76 31 00 02 a4 6d 61 70 31 d3 80 00 00 00 00 00 00 00 00 01 **c4 02** cc ab **c4 02** cc ef 00

Reply:
We'll want to replace this with our own/everparse framing anyway.